reposerver/repository/repository.go

```diff
@@ -1428,6 +1546,38 @@ func getResolvedRefValueFile(
 	return resolvedPath, nil
 }
 
+// getResolvedOCIRefValueFile handles OCI ref values by using the already extracted OCI content
+func getResolvedOCIRefValueFile(
+	rawValueFile string,
+	env *v1alpha1.Env,
+	allowedValueFilesSchemas []string,
+	refSourceRepo string,
+	ociPaths utilio.TempPaths,
+) (pathutil.ResolvedFilePath, error) {
+	pathStrings := strings.Split(rawValueFile, "/")
+
+	// Get the OCI path from the ociPaths
+	ociPath := ociPaths.GetPathIfExists(refSourceRepo)
+	if ociPath == "" {
+		return "", fmt.Errorf("failed to find OCI repo %q", refSourceRepo)
+	}
+
+	// Remove first segment (the ref variable name) and resolve the path
+	pathStrings[0] = "" // Remove first segment. It will be inserted by pathutil.ResolveValueFilePathOrUrl.
```

Comment:
should we guard `pathStrings[0] = ""` with a length check? 
an empty rawValueFile (or a leading slash) would make pathStrings would leave pathStrings empty and panic when we touch index 0?